src/hooks/use-filter-query.tsx

```diff
@@ -24,26 +24,27 @@ export const useFilterQuery = <T extends object>({
 }: UseFilterQueryOptions<T>): FilterQueryHook<T> => {
   const [searchParams, setSearchParams] = useSearchParams()
   const parsedFilters = parseQueryParams(searchParams, defaultFilters)
-  const [filters, setFilters] = useState<T>({
-    ...defaultFilters,
-    ...parsedFilters
-  })
 
-  const updateFilter = useCallback(<K extends keyof T>(value: T[K], key: K) => {
-    setFilters((prevFilters) => ({ ...prevFilters, [key]: value }))
-  }, [])
+  const filters = useMemo(
+    () => ({ ...defaultFilters, ...parsedFilters }),
+    [defaultFilters, parsedFilters]
+  )
 
-  const updateFilterInQuery = useCallback(
-    <K extends keyof T>(value: T[K], key: K) => {
-      updateFilter(value, key)
-      if (value) {
-        searchParams.set(String(key), String(value))
-      } else {
-        searchParams.delete(String(key))
+  const updateFiltersInQuery: UpdateFiltersInQuery<T> = useCallback(
+    (valuesToUpdate) => {
+      for (const key in valuesToUpdate) {
+        const value = valuesToUpdate[key]!
+
+        if (value) {
+          searchParams.set(String(key), String(value))
+        } else {
+          searchParams.delete(String(key))
+        }
```

Comment:
```suggestion
      valuesToUpdate.forEach((key) => {...})
```
It's better to use forEach here for better code readability.

Reply:
Sorry, cannot use `foreach` here because `valuesToUpdate` in an object. I think it will be cleaner and more performant because if we use `Object.keys()`, it first compute keys and then we should iterate over them again. So we iterate over keys two times